app/commands/v2/discard_draft.rb

```diff
@@ -13,8 +13,18 @@ def call
         increment_live_lock_version if live
 
         after_transaction_commit do
-          delete_draft_from_draft_content_store(draft_path)
-          send_live_to_draft_content_store(live)
+          if live
+            live_path = Location.find_by!(content_item: live).base_path
+
+            if live_path == draft_path
```

Comment:
Both branches of that `if` send the live version. I was suggesting...
```
if live
  live_path = Location.find_by!(content_item: live).base_path
  send_live_to_draft_content_store(live)
  delete_draft_from_draft_content_store(draft_path) if live_path != draft_path
else
  delete_draft_from_draft_content_store(draft_path)
end
```

I just find nested `if`s  make you have to pause and jump back a bit when you're reading the code so I try to avoid them.